File: packages/serverless/lib/plugins/aws/bedrock-agentcore/examples/python/langgraph-basic-docker/test-invoke.py
#!/usr/bin/env python3
"""
Test script to invoke the AgentCore Runtime using the correct AWS API.

Usage:
  RUNTIME_ARN=arn:aws:bedrock-agentcore:... python3 test-invoke.py

Or set RUNTIME_ARN environment variable before running.
"""
import boto3
import json
import os
import sys
import uuid

# Configuration - read from environment or use placeholder
RUNTIME_ARN = os.environ.get('RUNTIME_ARN')
REGION = os.environ.get('AWS_REGION', 'us-east-1')
SESSION_ID = str(uuid.uuid4())  # Generate a unique session ID

if not RUNTIME_ARN:
    print("Error: RUNTIME_ARN environment variable is required.")
    print("Usage: RUNTIME_ARN=<your-runtime-arn> python3 test-invoke.py")
    print("\nGet your runtime ARN from: serverless info")
    sys.exit(1)

def invoke_agent(input_text: str):
    """Invoke the AgentCore Runtime agent."""
    print(f"🤖 Invoking agent with input: '{input_text}'")
    print(f"📍 Runtime ARN: {RUNTIME_ARN}")
    print(f"🔑 Session ID: {SESSION_ID}\n")

    try:
        # Use bedrock-agentcore client
        client = boto3.client('bedrock-agentcore', region_name=REGION)

        # Prepare the payload
        payload = json.dumps({"prompt": input_text}).encode()

        # Invoke the agent runtime
        response = client.invoke_agent_runtime(
            agentRuntimeArn=RUNTIME_ARN,
            runtimeSessionId=SESSION_ID,
            payload=payload
        )

        print("✅ Response received:")
        print("-" * 50)

        # Handle streaming response
        if "text/event-stream" in response.get("contentType", ""):
            content = []
            for line in response["response"].iter_lines(chunk_size=10):
                if line:
                    line = line.decode("utf-8")
                    if line.startswith("data: "):
                        line = line[6:]
                    print(line)
                    content.append(line)
            print("\nComplete response:", "\n".join(content))
        elif response.get("contentType") == "application/json":
            # Handle standard JSON response
            content = []
            for chunk in response.get("response", []):
                content.append(chunk.decode('utf-8'))
            print(json.loads(''.join(content)))
        else:
            # Print raw response for other content types
            print(response)

        print("\n" + "-" * 50)
        print("✅ Invocation completed successfully!")

    except Exception as e:
        print(f"❌ Error invoking agent: {e}")
        sys.exit(1)

if __name__ == '__main__':
    # Test with a simple calculation
    invoke_agent("What is 25 multiplied by 4?")

    print("\n" + "=" * 50)
    print("\n🧪 Second test: Time query\n")

    # Test with time query
    invoke_agent("What is the current time?")

File: packages/serverless/lib/plugins/aws/bedrock-agentcore/examples/javascript/langgraph-streaming/test-invoke.js
#!/usr/bin/env node
/**
 * Test script to invoke the LangGraph streaming agent deployed to AgentCore Runtime.
 *
 * This script handles SSE (Server-Sent Events) streaming responses,
 * printing each token as it arrives for real-time output.
 *
 * Usage:
 *   RUNTIME_ARN=arn:aws:bedrock-agentcore:... node test-invoke.js
 *
 * Or set RUNTIME_ARN environment variable before running.
 *
 * Requires: @aws-sdk/client-bedrock-agentcore
 *   npm install @aws-sdk/client-bedrock-agentcore
 */

import {
  BedrockAgentCoreClient,
  InvokeAgentRuntimeCommand,
} from '@aws-sdk/client-bedrock-agentcore'
import { randomUUID } from 'node:crypto'

const RUNTIME_ARN = process.env.RUNTIME_ARN
const REGION = process.env.AWS_REGION || 'us-east-1'
const SESSION_ID = randomUUID()

if (!RUNTIME_ARN) {
  console.error('Error: RUNTIME_ARN environment variable is required.')
  console.error('Usage: RUNTIME_ARN=<your-runtime-arn> node test-invoke.js')
  console.error('\nGet your runtime ARN from: serverless info')
  process.exit(1)
}

const client = new BedrockAgentCoreClient({ region: REGION })

async function invokeAgent(inputText) {
  console.log(`Invoking agent with input: '${inputText}'`)
  console.log(`Runtime ARN: ${RUNTIME_ARN}`)
  console.log(`Session ID: ${SESSION_ID}\n`)

  const command = new InvokeAgentRuntimeCommand({
    agentRuntimeArn: RUNTIME_ARN,
    runtimeSessionId: SESSION_ID,
    payload: Buffer.from(JSON.stringify({ prompt: inputText })),
    contentType: 'application/json',
    accept: 'application/json, text/event-stream',
  })

  const response = await client.send(command)

  console.log('Response received (streaming):')
  console.log('-'.repeat(50))

  const contentType = response.contentType || ''
  const body = response.response

  if (contentType.includes('text/event-stream')) {
    /**
     * Handle SSE streaming response.
     * Each chunk is an SSE-formatted data line: "data: ...\n\n"
     */
    const chunks = []
    for await (const chunk of body) {
      const text =
        typeof chunk === 'string' ? chunk : new TextDecoder().decode(chunk)

      /**
       * Parse SSE lines and extract data payloads
       */
      const lines = text.split('\n')
      for (const line of lines) {
        if (line.startsWith('data: ')) {
          const data = line.slice(6)
          if (data && data !== '[DONE]') {
            try {
              const parsed = JSON.parse(data)
              if (typeof parsed === 'string') {
                process.stdout.write(parsed)
                chunks.push(parsed)
              } else if (parsed.text) {
                process.stdout.write(parsed.text)
                chunks.push(parsed.text)
              } else if (parsed.data?.text) {
                process.stdout.write(parsed.data.text)
                chunks.push(parsed.data.text)
              }
            } catch {
              /**
               * Not JSON — treat as plain text chunk
               */
              process.stdout.write(data)
              chunks.push(data)
            }
          }
        }
      }
    }
    console.log('\n')
  } else if (typeof body === 'string') {
    console.log(body)
  } else if (body && typeof body[Symbol.asyncIterator] === 'function') {
    const chunks = []
    for await (const chunk of body) {
      chunks.push(
        typeof chunk === 'string' ? chunk : new TextDecoder().decode(chunk),
      )
    }
    const text = chunks.join('')
    try {
      console.log(JSON.stringify(JSON.parse(text), null, 2))
    } catch {
      console.log(text)
    }
  } else if (body && typeof body.read === 'function') {
    const data = await body.read()
    const text = new TextDecoder().decode(data)
    try {
      console.log(JSON.stringify(JSON.parse(text), null, 2))
    } catch {
      console.log(text)
    }
  } else {
    console.log(response)
  }

  console.log('-'.repeat(50))
  console.log('Invocation completed!\n')
}

async function main() {
  console.log('LangGraph Streaming Agent Test Suite')
  console.log('='.repeat(50) + '\n')

  /**
   * Test 1: Simple calculation — triggers tool use, then streams the response
   */
  console.log('Test 1: Calculator (with tool call + streamed response)\n')
  await invokeAgent('What is 25 multiplied by 4?')

  console.log('='.repeat(50))

  /**
   * Test 2: Conversational — streams tokens directly without tool calls
   */
  console.log('\nTest 2: Conversational (pure streamed text)\n')
  await invokeAgent('Write a short haiku about streaming data in real time.')

  console.log('='.repeat(50))

  /**
   * Test 3: Time query — tool call followed by streamed explanation
   */
  console.log('\nTest 3: Time query (tool call + streamed response)\n')
  await invokeAgent('What is the current time in New York?')

  console.log('='.repeat(50))
  console.log('All tests completed!')
}

main().catch((err) => {
  console.error('Test failed:', err)
  process.exit(1)
})
